app/controllers/course/assessment/assessments_controller.rb

```diff
@@ -43,6 +43,21 @@ def destroy
     end
   end
 
+  # Reorder questions for an assessment
+  def reorder
+    unless valid_ordering?(question_order_params)
+      raise ArgumentError, 'Invalid ordering for assessment questions'
+    end
+
+    question_order_params.each_with_index do |id, index|
+      questions_hash[id].weight = index
+    end
+
+    Course::Assessment::Question.transaction do
+      questions_hash.values.map(&:save)
```

Comment:
`save` will fail silently and won't rollback the transaction, use `save!`.

Or would be better to use `update_column`, save triggers all validations which are costly.